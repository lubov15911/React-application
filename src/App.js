import React from 'react';

import { SearchPanel } from './SearchPanel.js'; // Functional component + Pure Component
import { SearchResults } from './SearchResults.js'; // Component

import './App.scss';

export let App = React.createElement(
    'div',
    {
        className: 'app'
    },
    <h1> Hello, World! </h1>,
    <SearchPanel />,
    <SearchResults />
);
