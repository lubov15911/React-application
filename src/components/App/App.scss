body {
  background: #555;
  font-size: 10px;
  font-family: Arial, Helvetica, sans-serif;
  margin: 5.5em;
}

p {
  margin: 0;
}

.app {
  background: #232323;
  box-shadow: 0 0.2rem 3.4rem 0 rgba(0, 0, 0, 0.27);
  color: white;
  width: 120em;
  margin: 0 auto;
}
