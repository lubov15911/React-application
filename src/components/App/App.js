import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import SearchPage from '../SearchPage';
import MoviePage from '../MoviePage';
import ErrorBoundary from '../ErrorBoundary';

import './App.scss';

const propTypes = {
    hasMovieData: PropTypes.bool.isRequired,
};

const App = ({ hasMovieData }) => (
    <ErrorBoundary>
        <div className="app">
            {hasMovieData ? <MoviePage /> : <SearchPage />}
        </div>
    </ErrorBoundary>
);
App.propTypes = propTypes;

const mapStateToProps = (state) => {
    return {
        hasMovieData: !!state.movieData,
    }
};

export default connect(mapStateToProps)(App);
