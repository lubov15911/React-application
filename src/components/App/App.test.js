import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import App from './App';

describe('App', () => {
    it('should render correctly without movieData', () => {
        const mockStore = configureStore()({
            movieData: null,
        });

        const component = shallow(<App store={mockStore} />).dive();

        expect(component).toMatchSnapshot();
    });

    it('should render correctly with movieData', () => {
        const mockStore = configureStore()({
            movieData: {},
        });

        const component = shallow(<App store={mockStore} />).dive();

        expect(component).toMatchSnapshot();
    });
});
