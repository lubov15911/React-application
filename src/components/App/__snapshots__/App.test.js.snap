// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly with movieData 1`] = `
<ErrorBoundary>
  <div
    className="app"
  >
    <Connect(MoviePage) />
  </div>
</ErrorBoundary>
`;

exports[`App should render correctly without movieData 1`] = `
<ErrorBoundary>
  <div
    className="app"
  >
    <Connect(SearchPage) />
  </div>
</ErrorBoundary>
`;
